### PR TITLE
Finalisation of Update Notification Function

### PR DESCRIPTION
**Pull request by MarkSummerville**

### CODE REVIEW

File: TVRename#/Utility/Helpers.cs [hunk 1]
```diff
@@ -659,5 +659,41 @@ public static string RemoveDiacritics(string stIn)
...
+
+        internal static bool GreaterVersionString(string versionNumber1, string versionNumber2)
+        {
+            //Returns true iff versionNumber 1 > versionNumber2
+
+            //First check to see that they both have text
+            if (String.IsNullOrWhiteSpace(versionNumber2)) return true;
+            if (String.IsNullOrWhiteSpace(versionNumber1)) return false;
+
+            char[] endings = new char[] {'-', ' '};
```
Comment (JoeBiellik) [inline TVRename#/Utility/Helpers.cs]: A git tag cannot have a space so I don't think the check is needed.

File: TVRename#/Utility/Helpers.cs [hunk 2]
```diff
@@ -659,5 +659,41 @@ public static string RemoveDiacritics(string stIn)
...
+            if (versionNumber2.LastIndexOfAny(endings) == -1) return false;
+
+            //We have 1 suffixes
+            string suffix1 = versionNumber1.Substring(versionNumber1.LastIndexOfAny(endings)+1);
+            string suffix2 = versionNumber2.Substring(versionNumber2.LastIndexOfAny(endings)+1);
+
+            //Compare alphabetically alpha1 < alpha2 < beta1 < beta2 < rc1 < rc2 etc
+            return (String.Compare(suffix1, suffix2, StringComparison.Ordinal) > 0);
+
+        }
```
[the remark this answers is not in the document]
Comment (JoeBiellik) [reply TVRename#/Utility/Helpers.cs]: This looks good if you don't want to use regex or a library.

File: TVRename#/Utility/Helpers.cs (hunk 1 above)
Comment (MarkSummerville) [reply TVRename#/Utility/Helpers.cs]: I'm using the same method to compare the user's display version to the github tag though. I thought there was something funky that we do to convert x.y.x-rc2 to 'x.y.z rc2'?

Comment (JoeBiellik) [reply TVRename#/Utility/Helpers.cs]: Good point, [AppVeyor formats](https://github.com/TV-Rename/tvrename/blob/master/appveyor.yml#L17-L28) the version number.

Comment (MarkSummerville) [reply TVRename#/Utility/Helpers.cs]: Looking at the code then I have not taken into account `-replace "([a-z]+)(\d+)",'$1 $2'` , so we may get some erroneous update messages on the beta branch.

Could we have AppVeyor 'inject' the tagged version name into `AssemblyInfo.cs` (in the same way it does for the pretty-printed version) and then have a helper method expose it? That way I do not have to revert the pretty-printed version back to the 'tag' form to compare?

Comment (JoeBiellik) [reply TVRename#/Utility/Helpers.cs]: No, half the pain of getting it working for the builds was that Windows validates and enforces z.y.x.w format, no alpha characters, for version numbers. AppVeyor is already embedding the "friendly" version in the only available area that Windows does not validate. The validated fields are populated as best as they can be, 2.3.0.0 for 2.3.0 RC 2, etc. I don't see an easy way of doing this without coming up with a different method for automated builds.

At this point I would use regex:
```cs
public class SemVer
{
    public int Major { get; set; }
    public int Minor { get; set; }
    public int Patch { get; set; }
    public string Prerelease { get; set; }
    public string Build { get; set; }

    public static SemVer FromSemantic(string version)
    {
        if (string.IsNullOrWhiteSpace(version)) throw new ArgumentException("The provided version string is invalid.", nameof(version));

        Regex regex = new Regex(@"^(?<major>[0-9]+)((\.(?<minor>[0-9]+))(\.(?<patch>[0-9]+))?)?(\-(?<pre>[0-9A-Za-z\-\.]+|[*]))?(\+(?<build>[0-9A-Za-z\-\.]+|[*]))?$", RegexOptions.ExplicitCapture);

        Match match = regex.Match(version);

        if (!match.Success || !match.Groups["major"].Success || !match.Groups["minor"].Success || !match.Groups["patch"].Success) throw new ArgumentException("The provided version string is invalid.", nameof(version));

        return new SemVer
        {
            Major = int.Parse(match.Groups["major"].Value),
            Minor = int.Parse(match.Groups["minor"].Value),
            Patch = int.Parse(match.Groups["patch"].Value),
            Prerelease = match.Groups["pre"].Value,
            Build = match.Groups["build"].Value
        };
    }

    public static SemVer FromFriendly(string version)
    {
        if (string.IsNullOrWhiteSpace(version)) throw new ArgumentException("The provided version string is invalid.", nameof(version));

        Regex regex = new Regex(@"^(?<major>[0-9]+)((\.(?<minor>[0-9]+))(\.(?<patch>[0-9]+))?)?( (?<pre>[0-9A-Za-z\- ]+))?$", RegexOptions.ExplicitCapture);

        Match match = regex.Match(version);

        if (!match.Success || !match.Groups["major"].Success || !match.Groups["minor"].Success) throw new ArgumentException("The provided version string is invalid.", nameof(version));

        return new SemVer
        {
            Major = int.Parse(match.Groups["major"].Value),
            Minor = int.Parse(match.Groups["minor"].Value),
            Patch = match.Groups["patch"].Success ? int.Parse(match.Groups["patch"].Value) : 0,
            Prerelease = match.Groups["pre"].Value.Replace(" ", string.Empty)
        };
    }
}
```

Comment (MarkSummerville) [reply TVRename#/Utility/Helpers.cs]: Great - thanks very much - I will integrate tonight

Cheers

**Pull request merged**